src/shogun/mathematics/linalg/SGLinalg.cpp

```diff
@@ -113,13 +113,37 @@ T SGLinalg::dot(BaseVector<T> *a, BaseVector<T> *b) const
 	}
 }
 
+template <class T>
+T SGLinalg::sum(BaseVector<T> *vec) const
+{
+	if (vec->onGPU())
+	{
+		if (this->hasGPUBackend())
+		{
+			return m_gpubackend->sum<T>(static_cast<GPUVector<T>&>(*vec));
+		}
+		else
+		{
+			SG_SERROR("User did not register GPU backend. \n");
```

Comment:
Why was there an error message in the sum implementation if there is already one here?

Also, can we just call some method that throws the error? We dont want to re-write this all the time.
Finally, can you please provide a better error message? Like "Tried to call GPU method %s without registering a GPU backend first. Register as blablabla"